### ISSUE TITLE
FR: report on misconfigured vlan info

### ISSUE THREAD

**Issue by kingtrw** [netdisco/netdisco#1022]
I've discovered that if you're setting vlan memberships on a trunk by hand, on some switches it's possible to specify a vlan on a trunk that does not exist in the switch's vlan database.

We seem to have a lot of these resulting from a trunk being specified with "switchport trunk allowed vlan X,Y,Z" and someone doing 'no vlan X'.

Is there any way a report could be made to identify:

* vlans that are specified on a switchport that don't exist in the switch's vlan database
* vlans that are specified in the switch's vlan database that aren't assigned to any switchports

It would be very handy!

**Comment by ollyg** [comment]
hmm well the issue @kingtrw is that the vlan database concept is proprietary, I think. To my knowledge, devices don't really care about vlan assignment, they just get the tagged frames and send them. The database is some extra sugar from the vendor, to support your management.

so we would need to gather the database info, and store it, and we don't have a slot in the database to do this (yet).

going to close and suggest one for wishlist?

**Comment by kingtrw** [comment]
I've done a bit of testing and I think the second one of these reports should be possible with the info netdisco already has.

I've created a vlan 666 that's not assigned to any ports. ND can see the vlan and name in the VLANs tab - and it can see that it's not assigned to any ports, so I was hoping it would be possible for a report that could say "Hey, this switch has vlan X, but it's not on any ports".


I'm still investigating the situation that (apparently) allowed autonegotiation to make a vlan appear on a port when it had been removed from the switch - so the that report request can be ignored for now as it's not viable.
